File: KevinRene/src/cons/ql/ast/Binary.java
package cons.ql.ast;

public abstract class Binary extends Expr {
	
	protected Expr left, right;
	protected String operator;
	
	public Binary(Expr left, Expr right, String operator) {
		this.left = left;
		this.right = right;
		this.operator = operator;
		
		System.out.println(show());
	}

	public String show() {
		return this.left.show() + " " + this.operator + " " + this.right.show();
	}
}

File: FelipezConde/QuestionnaireLanguage/AST/Nodes/Expressions/Binary.cs
using AST.Nodes.Interfaces;

namespace AST.Nodes.Expressions
{
    public abstract class BaseBinary : BaseExpression
    {
        private readonly BaseExpression left; 
        private readonly BaseExpression right;

        protected BaseBinary(BaseExpression left, BaseExpression right, PositionInText pos)
            :base(pos)
        {
            this.left = left; 
            this.right = right;
        }

        public BaseExpression Left()
        { return left; }

        public BaseExpression Right()
        { return right; }

    }
}
